tests/measurements/test_purity_measurement.py

```diff
@@ -0,0 +1,374 @@
+        assert qml.math.allclose(purity, expected_purity)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("device", grad_supported_devices)
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wires,is_partial", wires_list)
+    @pytest.mark.parametrize("diff_method", diff_methods)
+    def test_IsingXX_qnode_purity_grad_torch(self, device, param, wires, is_partial, diff_method):
+        """Test purity for a QNode gradient with torch."""
+
+        import torch
+
+        dev = qml.device(device, wires=2)
+
+        @qml.qnode(dev, interface="torch", diff_method=diff_method)
+        def circuit(x):
+            qml.IsingXX(x, wires=[0, 1])
+            return qml.purity(wires=wires)
+
+        expected_grad = expected_purity_grad_ising_xx(param) if is_partial else 0
+
+        param = torch.tensor(param, dtype=torch.float64, requires_grad=True)
+        purity = circuit(param)
+        purity.backward()
+        grad_purity = param.grad
+
+        assert qml.math.allclose(grad_purity, expected_grad, rtol=1e-04, atol=1e-05)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("device", devices)
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wires,is_partial", wires_list)
+    def test_IsingXX_qnode_purity_tf(self, device, param, wires, is_partial):
+        """Tests purity for a qnode"""
+
+        import tensorflow as tf
+
+        dev = qml.device(device, wires=2)
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(x):
+            qml.IsingXX(x, wires=[0, 1])
+            return qml.purity(wires=wires)
+
+        purity = circuit(tf.Variable(param))
+        expected_purity = expected_purity_ising_xx(param) if is_partial else 1
+        assert qml.math.allclose(purity, expected_purity)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("device", grad_supported_devices)
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wires,is_partial", wires_list)
+    @pytest.mark.parametrize("diff_method", diff_methods)
+    def test_IsingXX_qnode_purity_grad_tf(self, device, param, wires, is_partial, diff_method):
+        """Test purity for a QNode gradient with tf."""
+
+        import tensorflow as tf
+
+        dev = qml.device(device, wires=2)
+
+        @qml.qnode(dev, interface="tf", diff_method=diff_method)
+        def circuit(x):
+            qml.IsingXX(x, wires=[0, 1])
+            return qml.purity(wires=wires)
+
+        grad_expected_purity = expected_purity_grad_ising_xx(param) if is_partial else 0
+
+        param = tf.Variable(param)
+        with tf.GradientTape() as tape:
+            purity = circuit(param)
+
+        grad_purity = tape.gradient(purity, param)
+
+        assert qml.math.allclose(grad_purity, grad_expected_purity, rtol=1e-04, atol=1e-05)
+
+    @pytest.mark.parametrize("device", devices)
+    @pytest.mark.parametrize("return_new", return_new)
```

Comment:
I don't think we need to test this using the new return types, given that nothing changes.

Reply:
Same here, what lines are not covered if you remove this test?

Reply:
The shape_new method of the PurityMP class